### PR TITLE
Migrate DragonFly

### PR DESCRIPTION
**Pull request by naheedsa**

### CODE REVIEW

File: indi-armadillo-platypus/dragonfly_dome.cpp [hunk 1]
```diff
@@ -168,8 +168,8 @@ void DragonFlyDome::ISGetProperties(const char *dev)
 {
     INDI::Dome::ISGetProperties(dev);
 
-    defineProperty(&PerPortSP);
-    loadConfig(true, PerPortSP.name);
+    defineProperty(PerPortSP);
+    loadConfig(true, PerPortSP.getName());
```
Comment (knro) [inline indi-armadillo-platypus/dragonfly_dome.cpp:172]: no need to call loadConfig, you can call PerPortSP.load() but the right place to do this is not in ISGetProperties but in after line 148 directly where the property is initialized.

File: indi-armadillo-platypus/dragonfly_dome.cpp [hunk 2]
```diff
@@ -276,12 +276,12 @@ bool DragonFlyDome::ISNewSwitch(const char *dev, const char *name, ISState *stat
...
+        if (PerPortSP.isNameMatch(name))
         {
-            IUUpdateSwitch(&PerPortSP, states, names, n);
-            PerPortSP.s = IPS_OK;
-            IDSetSwitch(&PerPortSP, nullptr);
-            saveConfig(true, PerPortSP.name);
+            PerPortSP.update(states, names, n);
+            PerPortSP.setState(IPS_OK);
+            PerPortSP.apply();
+            saveConfig(true, PerPortSP.getName());
```
Comment (knro) [inline indi-armadillo-platypus/dragonfly_dome.cpp:284]: Use the new style. saveConfig(PerPortSP);

File: indi-armadillo-platypus/dragonfly_dome.cpp [hunk 3]
```diff
@@ -317,23 +317,23 @@ bool DragonFlyDome::ISNewNumber(const char *dev, const char *name, double values
...
+        if (DomeControlSensorNP.isNameMatch(name))
         {
-            IUUpdateNumber(&DomeControlSensorNP, values, names, n);
-            DomeControlSensorNP.s = IPS_OK;
-            IDSetNumber(&DomeControlSensorNP, nullptr);
-            saveConfig(true, DomeControlSensorNP.name);
+            DomeControlSensorNP.update(values, names, n);
+            DomeControlSensorNP.setState(IPS_OK);
+            DomeControlSensorNP.apply();
+            saveConfig(true, DomeControlSensorNP.getName());
```
Comment (knro) [inline indi-armadillo-platypus/dragonfly_dome.cpp:336]: Use new style

File: indi-armadillo-platypus/dragonfly_dome.cpp (hunk 1 above)
Comment (naheedsa) [reply indi-armadillo-platypus/dragonfly_dome.cpp]: sure, using now in line 157

**Pull request merged**